crypto/evp/evp_locl.h

```diff
@@ -95,3 +95,38 @@ void *evp_generic_fetch(OPENSSL_CTX *ctx, int operation_id,
                                             OSSL_PROVIDER *prov),
                         int (*up_ref_method)(void *),
                         void (*free_method)(void *));
+
+/* Helper functions to avoid duplicating code */
+
+/*
+ * The callbacks implement different ways to pass a params array to the
+ * provider.  They will return one of these values:
+ *
+ * -2 if the method doesn't come from a provider
```

Comment:
gaah, really want define's for these.

Reply:
Yeah well...  cleanup PR welcome

Reply:
... and then it sits around for weeks or months ;)  Been there, done that, have the t-shirt. Or at least the cleanup PR's (https://github.com/openssl/openssl/pulls/richsalz)

Reply:
I know the feeling.  But I find it utterly useless to make such a change in *one* spot and ignore all the others.  And this is not a cleanup PR.